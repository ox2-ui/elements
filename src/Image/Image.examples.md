### Image

    <Image
      customStyle={{width: '250px'}}
      value="http://imgur.com/7YG1YAU.png"
    />
