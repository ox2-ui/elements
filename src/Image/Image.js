import React, { PropTypes } from 'react';

const styles = {
  base: {
    display: 'block',
  },
};

/**
 * Image Component
 */
const Image = ({ className, style, value, customStyle }) => (
  <img
    className={className}
    src={value}
    style={{ ...styles.base, ...style, ...customStyle }}
  />
);


Image.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Image specific style properties.
   */
  customStyle: PropTypes.object.isRequired,
  /**
   * Override the inline-styles of the text element.
   */
  style: PropTypes.object,
  /**
   * Image data or url.
   */
  value: PropTypes.string.isRequired,
};

export default Image;
