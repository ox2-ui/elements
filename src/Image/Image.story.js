import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Image from './Image';

storiesOf('Image', module)
  .add('default', () => (
    <Image
      customStyle={{width: '250px'}}
      value="http://imgur.com/7YG1YAU.png"
    />
  ));
