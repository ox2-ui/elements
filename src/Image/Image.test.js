/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Image from './Image';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Image
      customStyle={{width: '250px'}}
      value="http://imgur.com/7YG1YAU.png"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
