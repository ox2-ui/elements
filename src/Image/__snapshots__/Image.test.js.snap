// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  customStyle={
    Object {
      "width": "250px",
    }
  }
  value="http://imgur.com/7YG1YAU.png"
>
  <img
    src="http://imgur.com/7YG1YAU.png"
    style={
      Object {
        "display": "block",
        "width": "250px",
      }
    }
  />
</Image>
`;
